### PR TITLE
Added code review process documentation.

### PR DESCRIPTION
**Pull request by joachimmetz**

### CODE REVIEW

**Comment by joachimmetz** [comment]
@Onager let's give reviewing documentation via github a try, maybe rietveld is less suitable for documentation reviews. We still should do a squash merge IMO.

Note that github review has a "[rich diff](https://github.com/log2timeline/l2tdocs/pull/1/files?short_path=552505a#diff-552505ae77fe5f5e2ba25f503ba282f0)" mode.

File: process/Code review process.asciidoc [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+named upload.py to upload code diffs to codereview.appspot.com. Though it has
+git support, Rietveld was build around svn and therefore does not entirely
+match a git-based workflow.
+
+Several utility scripts have been built around upload.py to make the code
+review process more user friendly.
+
+=== Need for change
+
+* current solution requires contributors to have commit access
```
Comment (Onager) [inline process/Code review process.asciidoc]: all contributors

File: process/Code review process.asciidoc [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+[preface]
+== Revision history
+[cols="1,1,1,5",options="header"]
+|===
+| Version | Author | Date | Comments
+| 0.0.1 | Joachim Metz | January 2016 | Initial version based on template.
+|===
+
+:numbered:
+== Introduction
```
Comment (Onager) [inline process/Code review process.asciidoc]: Move all this to the end, to a "background" section - most people won't care about why we do this, and just want to get to the how-to.

File: process/Code review process.asciidoc [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+....
+
+`4`. Make sure git is configured correctly:
+....
+git config --global user.name "Full Name"
+git config --global user.email name@example.com
+git config --global push.default matching
+....
+
+`5`. Make sure `.netrc` is configured
```
Comment (Onager) [inline process/Code review process.asciidoc]: Link to something - https://gist.github.com/technoweenie/1072829 ?

File: process/Code review process.asciidoc [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+=== Step 2 - Starting a new feature branch
```
Comment (Onager) [inline process/Code review process.asciidoc]: Start, not starting

File: process/Code review process.asciidoc [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+git checkout -b feature
+....
+
+`3.` Make the necessary code changes, commit them to your feature branch and upload them to your fork:
+....
+git push --set-upstream origin feature
+....
+
+=== Step 3 - Making changes
+Make your changes to the code and make sure to up them to the feature branch of
```
Comment (Onager) [inline process/Code review process.asciidoc]: +push

File: process/Code review process.asciidoc [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+your fork.
+
+Commit all code changes to your feature branch and upload them:
+....
+git push
+....
+
+=== Step 4 - Starting a code review
```
Comment (Onager) [inline process/Code review process.asciidoc]: Start

File: process/Code review process.asciidoc [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+
+* check the status of the pull request on the github project page e.g.
+https://github.com/log2timeline/plaso/pull/
+
+If tests are broken (and it is due to the CL) the reviewer will ask the
+contributor fix this.
+
+* comment on the changes on: https://codereview.appspot.com/[codereview.appspot.com]
+
+=== Step 6 - Updating a code review
```
Comment (Onager) [inline process/Code review process.asciidoc]: Update - all these headings need to be consistent.

File: process/Code review process.asciidoc [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+The process of updating a code reviews is very similar to that of starting a
```
Comment (Onager) [inline process/Code review process.asciidoc:156]: -a

File: process/Code review process.asciidoc [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+* checks if your local clone is up to date with the upstream/master and updates
+if necessary by running: `git pull --rebase upstream master`
+* uploads the diff with upstream/master to codereview.appspot.com
+* mails the maintainers
+
+`3.` Any update to the feature branch while there is a PR will trigger the
+automated testing.
+
+=== Step 7 - Merging the code review with upstream
+After the review is done the reviewer will merge the CL and submit it to upstream.
```
Comment (Onager) [inline process/Code review process.asciidoc:184]: Add a note that most committers won't need to do this step, just maintainers

File: process/Code review process.asciidoc (hunk 1 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: done

File: process/Code review process.asciidoc (hunk 2 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: Moved to appendix

File: process/Code review process.asciidoc (hunk 3 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: ack

File: process/Code review process.asciidoc (hunk 4 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: ack

File: process/Code review process.asciidoc (hunk 5 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: ack

File: process/Code review process.asciidoc (hunk 6 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: ack

File: process/Code review process.asciidoc (hunk 7 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: ack

File: process/Code review process.asciidoc (hunk 8 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc:156]: ack

File: process/Code review process.asciidoc (hunk 9 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc:184]: ack

File: process/Code review process.asciidoc [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+Copyright (C) 2016, log2timeline projects maintainers <log2timeline-maintainers@googlegroups.com>
+Permission is granted to copy, distribute and/or modify this document under the
+terms of the GNU Free Documentation License, Version 1.3 or any later version
+published by the Free Software Foundation; with no Invariant Sections, no
+Front-Cover Texts, and no Back-Cover Texts. A copy of the license is included
+in the "LICENSE" file.
+....
+
+:numbered:
+== Introduction
```
Comment (Onager) [inline process/Code review process.asciidoc]: I meant moving most the introduction, documentation of the old process and motivation to the end, and focusing on what we currently do right now.

**Comment by joachimmetz** [comment]
@Onager PTAL

File: process/Code review process.asciidoc [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+|===
+
+:numbered:
+== Introduction
+The plaso review process was in need of some revision to allow us to use the
+continuous integration (CI) testing functionality of github.
+
+== Code review process
+
+* Step 1 - set up your personal fork
```
Comment (Onager) [inline process/Code review process.asciidoc]: This bullet can go away now

File: process/Code review process.asciidoc [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+[cols="1,1,1,5",options="header"]
+|===
+| Version | Author | Date | Comments
+| 0.0.1 | Joachim Metz | January 2016 | Initial version based on template.
+|===
+
+:numbered:
+== Introduction
+The plaso review process was in need of some revision to allow us to use the
+continuous integration (CI) testing functionality of github.
```
Comment (Onager) [inline process/Code review process.asciidoc]: Move this to the appendix too, and just have: "The Plaso review process works as follows:" or similar.

**Comment by Onager** [comment]
Two small comments, LGTM

File: process/Code review process.asciidoc (hunk 12 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: ack

File: process/Code review process.asciidoc (hunk 11 above)
Comment (joachimmetz) [inline process/Code review process.asciidoc]: ack

**Pull request closed without merging**